src/net.cpp

```diff
@@ -1377,6 +1378,58 @@ void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_s
         if(FD_ISSET(hSocket, &fdsetError))
             error_set.insert(hSocket);
 }
+#else
+void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_set, std::set<SOCKET> &error_set)
+{
+    std::set<SOCKET> recv_select_set, send_select_set, error_select_set;
+    GenerateSelectSet(recv_select_set, send_select_set, error_select_set);
+
+    if (recv_select_set.empty() && send_select_set.empty() && error_select_set.empty()) {
+        interruptNet.sleep_for(std::chrono::milliseconds(SELECT_TIMEOUT_MILLISECONDS));
+        return;
+    }
+
+    std::vector<struct pollfd> pollfds;
+    pollfds.reserve(recv_select_set.size() + send_select_set.size() + error_select_set.size());
+
+    {
+        struct pollfd pollfd = {};
+
+        for (SOCKET hSocket : recv_select_set) {
+            pollfd.fd = hSocket;
+            pollfd.events = POLLIN;
+            pollfds.push_back(pollfd);
+        }
+
+        for (SOCKET hSocket : send_select_set) {
+            pollfd.fd = hSocket;
+            pollfd.events = POLLOUT;
+            pollfds.push_back(pollfd);
+        }
+
+        for (SOCKET hSocket : error_select_set) {
+            pollfd.fd = hSocket;
+            pollfd.events = POLLERR|POLLHUP;
```

Comment:
This is ignored and reported on `revents` in any case, no? I guess no harm in being explicit.

Reply:
I'm reusing the `struct pollfd` so it's necessary to overwrite the previous value.